nntrainer/layers/conv2d_layer.cpp

```diff
@@ -81,9 +78,13 @@ void Conv2DLayer::forwarding(sharedConstTensors in) {
   TensorDim &in_dim = input_dim[0];
   TensorDim &out_dim = output_dim[0];
 
-  TensorDim hidden_dim = output_dim[0];
   Tensor &hidden_ = net_hidden[0]->var;
-  hidden_.setZero();
+  if (hidden_.uninitialized()) {
```

Comment:
I wonder if we have to check this in every layer or just a specific case in conv2d?

Reply:
This is just from backward compatibility of hidden which was managed by layer.
I am not yet sure how graph is handling. 
Currently, net_buffer is under heavy updates - check #765, #766
Most likely this is redundant, and I will remove this later.

Reply:
Acked :)